### DISCUSSION TITLE
recovery/

### DISCUSSION THREAD

**Discussion by giscus** [zmike/zmike.github.io#56] [Announcements]
# recovery/

Struggling

https://www.supergoodcode.com/recovery/

**Comment by PeakKS** [comment]
Maybe instead of deprecating ExecuteIndirect Microsoft should have just deprecated the whole of D3D :P

**Reply by hammadnadeemx** [reply]
yeah and they should make Vulkan the standard API on windows !

**Comment by anon0824** [comment]
As a 3D engine developer, I am interested in Work Graphs, even after reading about its potentially dubious performance, mainly because it seems to be a programming model that is a good match for the node graphs given to artists in 3D modeling software.

For example, take Blender's Geometry Nodes. (It's a node graph for procedural geometry.)
In Blender, Geometry Nodes is implemented on the CPU, but what if we wanted to execute Geometry Nodes on the GPU instead?
Work Graphs seem like a great fit for such an implementation.
And, on the other hand, wouldn't it more of a pain to implement Geometry Nodes on the GPU without Work Graphs?

Bungie did a great talk about this sort of thing that you can read here (The Destiny Particle Architecture): https://advances.realtimerendering.com/s2017/index.html 
Their performant implementation in the final product entails compiling a separate shader for each individual particle system.
Might not Work Graphs, although perhaps at some cost to performance, make for a much more straightforward program, where instead of having a separate shader for each particle system generated at compile time, you can write a single shader in which each node in Geometry Nodes / a particle system's node graph is written as a single node-function in the shader?

As a shader programmer, which is preferrable-
debugging 100+ separate particle system / geometry node shaders generated from node graphs for a game,
or debugging a single shader's set of Work Graph nodes?
To me, the Work Graph seems like it would be much preferrable,
so much so that until Vulkan gets a vendor-agnostic Work Graphs extension, I might rather just implement such temporarily implement such features on the CPU for now.

**Comment by qwertychouskie** [comment]
New blog post when?  Not even a post about Vulkan 1.4, what a shame... ;)

**Comment by Permik** [comment]
![asking-for-supergoodchair](https://github.com/user-attachments/assets/1a158767-c3c8-4c45-870e-9d580c3bec68)
_Please stop cooking before the post gets burnt_ 🐸

**Reply by Permik** [reply]
Oh, I forgot you're on your va-cay, please ignore. Enjoy your holiday till the end and have a great new year!

**Comment by qwertychouskie** [comment]
Happy new year!  Can we expect more Superbly Excellent Software in the coming weeks?
